pennylane_qiskit/converter.py

```diff
@@ -59,19 +61,33 @@ def _check_circuit_and_bind_parameters(quantum_circuit: QuantumCircuit, params:
         params (dict): dictionary of the parameters in the circuit
 
     Returns:
-        qc (QuantumCircuit): quantum circuit with bound parameters
+        QuantumCircuit: quantum circuit with bound parameters
+        dict[qiskit.circuit.Parameter, pennylane.variable.VariableRef]: a dictionary mapping
+            qiskit parameters to PennyLane variables
+
     """
     if not isinstance(quantum_circuit, QuantumCircuit):
         raise ValueError("The circuit {} is not a valid Qiskit QuantumCircuit.".format(quantum_circuit))
 
     if params is None:
-        return quantum_circuit
+        return quantum_circuit, {}
+
+    # map qiskit parameters to PennyLane differentiable VariableRefs.
+    var_ref_map = {}
 
     for k, v in params.items():
-        if isinstance(v, qml.variable.Variable):
-            params.update({k: v.val})
+        # iterate through the parameters
+        if isinstance(v, qml.variable.VariableRef):
+            # If a parameter is a Variable reference, then it represents
+            # a differentiable parameter.
+            var_ref_map[k] = v
+
+    for k in var_ref_map:
+        # Since we cannot bind VariableRefs to Qiskit circuits,
+        # we must remove them from the binding dictionary before binding.
+        del params[k]
 
-    return quantum_circuit.bind_parameters(params)
+    return quantum_circuit.bind_parameters(params), var_ref_map
```

Comment:
It might be slightly unintuitive that here there's a tuple that is returned (but this is also connected to the original implementation which was my doing :sweat_smile: ). Given the change there seem to be four parts of logic here inside of this single function:
* check_circuit
* extract `VariableRefs`
* remove `VariableRefs` from params
* bind parameters

While in theory, this means four separate functions it might be worth having a separate function that solely extracts the `VariableRefs` and then later the `bind_parameters` function could delete the needed refs. To me, this latter separation would make this part easier to parse :)